File: single-language-analysis/dataset-files/python/test_smell_025.py
import pytest

class TestMovetest:
    def test_smell_025(self):
        g = Game(False)
        b = g.get_my_board()
        assert b is not None
        try:
            b.make_starting_board()
            assert b is not None
        except Exception as e:
            pytest.fail(str(e))

        g.set_current_player(ai_color)
        b.my_dice.roll(1, 1)
        assert b.my_dice.get_doublet_moves_countdown() == 4

        sg = StartGameStrategy()
        best = sg.pick_best_move(b, ai_color)
        print(best)
        assert best.is_possible()
        partials = best.get_my_partials()
        assert len(partials) == 4

        best.do_move()
        assert b.my_dice.get_doublet_moves_countdown() == 0

File: single-language-analysis/dataset-files/java/TestSmell025.java
// Original URL: https://github.com/backgammon-java-ai/backg_java2012/blob/ca26b37a9a649d3ed5eb97cfceb9e5719b269bb7/MoveTest.java#L49-L78

public class TestSmell025 {
    @Test
    public void testMove() {
        /* old test, not working anymore, maybe it needs specific board
           probably has issue with the game needing a first roll of dice
           to choose a currentPlayer */
        g = new Game(false);
        b = g.getMyBoard( );
        assertNotNull(b);
        try {
            b.makeStartingBoard( );/* regular game */
            assertNotNull(b);
        } catch(Exception e) {
            /* isn't there a way to test without catching exceptions? */
            fail(e.toString( ));
        }

        g.setCurrentPlayer(aiColor);
        b.myDice.roll(1,1); /* alternative syntax:b1.myDice.roll(1,2) */
        assertEquals(4, b.myDice.getDoubletMovesCountdown( ) );

        StartGameStrategy sg = new StartGameStrategy( );
        Move best = sg.pickBestMove(b,aiColor);
        System.out.println(best);
        assertTrue( best.isPossible( ) );
        ArrayList<PartialMove> partials = best.getMyPartials();
        assertEquals(4, partials.size( ) );

        best.doMove( );
        assertEquals(0, b.myDice.getDoubletMovesCountdown( ) );
    } /* testMove( ) */
}
